### PR TITLE
Refactor select to typescript

### PR DESCRIPTION
**Pull request by gonchihernandez**
#### :page_facing_up: Description:

This PR converts the Select component to TypeScript. To keep consistence in the theme I followed an other branch as an example.

---

#### :heavy_check_mark: Tasks:

* Define  Option interface
* Define Select interface

---

@loopstudio/react-devs

### CODE REVIEW

File: src/features/app/components/Form/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { SelectHTMLAttributes } from 'react';
+
+import PropTypes from 'prop-types';
```
Comment (Tiago-Zarzavidjian) [inline src/features/app/components/Form/Select.tsx:3]: There is no need for that empty line.

File: src/features/app/components/Form/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  name: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  label: PropTypes.string,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      label: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+};
```
Comment (Tiago-Zarzavidjian) [inline src/features/app/components/Form/Select.tsx:67]: As we are using TypeScript now, you should remove everything related to `PropTypes`.

**Comment by Tiago-Zarzavidjian** [comment]
Please remove all the `.js` files and don't forget to migrate the component styles as well.

**Review by nicocadq — COMMENTED** [review]
Delete the `js` file

**Comment by elestu** [comment]
@gonchihernandez 
- Remove `index.js`
- Check that old js file has been removed

**Review by elestu — APPROVED** [review]

**Pull request merged**